### PR TITLE
fix: Read receipts not generated when responding from notification (WPB-8756)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8756" title="WPB-8756" target="_blank">BugWPB-8756</a>  [Android] Read receipts not generated when responding from notification popup
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When we reply from head up notification, we don't mark older messages as read

### Solutions

Call `updateConversationReadDateUseCase` when replying from notifications

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/NotificationReplyReceiver.kt [hunk 1]
```diff
@@ -63,11 +64,19 @@ class NotificationReplyReceiver : BroadcastReceiver() { // requires zero argumen
...
-                            { updateNotification(context, conversationId, qualifiedUserId, replyText) }
+                    launch {
+                        messages.sendTextMessage(qualifiedConversationId, replyText)
+                            .fold(
+                                { updateNotification(context, conversationId, qualifiedUserId, null) },
+                                { updateNotification(context, conversationId, qualifiedUserId, replyText) }
+                            )
+                    }
+                    launch {
+                        conversations.updateConversationReadDateUseCase(
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/NotificationReplyReceiver.kt:75]: `updateConversationReadDateUseCase` will only send last read event to self clients and not send any read receipt

**Review by MohamadJaara — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/NotificationReplyReceiver.kt (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/NotificationReplyReceiver.kt:75]: my bad it does send it, it was just not clear

**Review by MohamadJaara — APPROVED** [review]

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
the goeal is to "mark only the last x number of messages that are displayed currently in the notification as read but leave the conversation as not read?" and `updateConversationReadDateUseCase` will also mark the conversation as read for self user

**Comment by ohassine** [comment]
> the goeal is to "mark only the last x number of messages that are displayed currently in the notification as read but leave the conversation as not read?" and `updateConversationReadDateUseCase` will also mark the conversation as read for self user

But why ? if it's marked as read for other users why we keep it unread for self user ? it will be somehow confusing

**Review by borichellow — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Review by MohamadJaara — APPROVED** [review]

**Pull request merged**